flow/aggregates.py

```diff
@@ -0,0 +1,463 @@
+        from an aggregator function to a subset of jobs as tuple.
+        """
+        nested_aggregates = []
+
+        for aggregate in aggregated_jobs:
+            try:
+                filter_aggregate = []
+                for job in aggregate:
+                    if job is None:
+                        continue
+                    elif job not in project:
+                        raise LookupError(f'The signac job {job.get_id()} not found in {project}')
+                    filter_aggregate.append(job)
+                filter_aggregate = tuple(filter_aggregate)
+                # Store aggregate in this instance
+                nested_aggregates.append(filter_aggregate)
+                # Store aggregate by their ids in order to search through id
+                self._aggregate_ids[get_aggregate_id(filter_aggregate)] = filter_aggregate
+            except TypeError:  # aggregate is not iterable
+                raise ValueError("Invalid aggregator function provided by "
+                                 "the user.")
+
+        self._aggregates = nested_aggregates
+
+
+class _DefaultAggregateStore:
+    """This class holds the information of the project associated with
+    an operation function aggregated by the default aggregates i.e.
+    :py:class:`Aggregate.groupsof(1)`.
+
+    When iterated through it's instance, it yields a tuple of a single job from
+    the Project.
+
+    :param jobs:
+        The signac job handles
+    :type jobs:
+        Iterator
+    """
+    def __init__(self, jobs):
+        self._project = None
+        self._register_project(jobs)
+
+    def __iter__(self):
+        for job in self._project:
+            yield (job,)
+
+    def __getitem__(self, id):
+        "Return a tuple of a single job via job id"
+        try:
+            return (self._project.open_job(id=id),)
+        except KeyError:
+            raise LookupError(f"Did not find job with id {id}.")
+
+    def __contains__(self, job):
+        """Return whether the job is present in the project associated with this
+        instance of :py:class:`_DefaultAggregateStore`
+        """
+        if job in self._project:
+            return True
+        else:
+            return False
+
+    def __len__(self):
+        return len(self._project)
+
+    def __eq__(self, other):
+        return type(self) == type(other) and \
+               self._project == other._project
+
+    def __hash__(self):
+        blob = str(self._project)
+        return int(sha1(blob.encode('utf-8')).hexdigest(), 16)
+
+    def _generate_aggregates(self, jobs):
+        """We have to store self._project when this method is invoked
+        This is because we will then iterate over that project in
+        order to return an aggregates of one.
+        """
+        self._register_project(jobs)
+
+    def _register_project(self, jobs):
+        if isinstance(jobs, list):  # Got a list of jobs
+            self._project = jobs[0]._project
```

Comment:
This is internal, should we ever be passed a list of jobs rather than a `FlowProject`? Considering we control this.